### PR TITLE
refactor: finalize initial generation

### PR DESCRIPTION
**Pull request by Lodin**
* Use Vite instead of web-test-runner.
* Use existing web-types.json descriptions instead of web-component-analyzer.
* Add fine-grained entrypoints for components.
* Export all components from root for better picking.
* Add script for generating entrypoint stubs.
* Add script for generating web-types.json schema typings.
* Add script for updating package.json's "exports" section.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,105 +5,322 @@
...
     "react-dom": "^18.2.0",
     "rimraf": "^3.0.2",
-    "ts-node": "^10.9.1",
-    "typescript": "^4.8.0",
-    "web-component-analyzer": "^2.0.0-next.4"
+    "rollup-plugin-esbuild": "^4.10.1",
+    "typescript": "^4.8.4",
+    "vite": "^3.1.8"
+  },
+  "exports": {
```
[the remark this answers is not in the document]
Comment (platosha) [reply package.json:99]: I guess having the real files in the `dist` directory could be one advantage, but seems that you’re not going that way...

File: tsconfig.build.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "extends": "./tsconfig.json",
+  "include": ["src"],
+  "exclude": ["node_modules", "dist"]
```
Comment (platosha) [inline tsconfig.build.json:4]: Perhaps you want to get rid if the `dist` directory here.

File: utils/commonjsFix.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+// From https://github.com/modernweb-dev/web/issues/1700 -> https://gist.github.com/lucaelin/ffa48cba2253e9ad662b71e38444a135
```
Comment (platosha) [inline utils/commonjsFix.js:1]: Is this still used?

**Review by platosha — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lodin) [reply package.json:100]: Added a `dist` folder instead

**Review by platosha — APPROVED** [review]

**Review by platosha — APPROVED** [review]

**Pull request merged**